src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -102,13 +111,228 @@ private static async Task<VersionStamp> GetDocumentSemanticVersionAsync(Document
             return semanticVersion;
         }
 
-        // Internal for testing
-        internal async Task<SemanticTokens?> GetSemanticTokensAsync(
+        private async Task<SemanticTokens?> GetCachedTokensAsync(
             TextDocumentIdentifier textDocumentIdentifier,
             DocumentSnapshot documentSnapshot,
             int documentVersion,
             VersionStamp semanticVersion,
             Range range,
+            ILogger logger,
```

Comment:
Do we change which loggers get passed through here by chance? Wondering if the `_logger` member isn't sufficient